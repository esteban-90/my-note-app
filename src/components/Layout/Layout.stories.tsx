import type { ComponentStory, ComponentMeta } from '@storybook/react'
import { Layout } from './Layout'

export default {
  title: 'Components/Layout',
  component: Layout,
} as ComponentMeta<typeof Layout>

const Template: ComponentStory<typeof Layout> = (args) => <Layout {...args} />

export const LayoutExample = Template.bind({})
