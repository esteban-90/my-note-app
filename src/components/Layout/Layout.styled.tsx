import styled from '@emotion/styled'
import { bgTransition, centered } from '@/styles'

export const Wrapper = styled.div`
  width: 100%;
  height: 100vh;
  color: var(--color-text);
  background-color: var(--color-bg);
  ${centered}
  ${bgTransition}
`

export const Content = styled.div`
  width: 100%;
  max-width: 480px;
  height: 90vh;
  background-color: var(--color-white);
  box-shadow: 1px 1px 6px rgba(0, 0, 0, 0.05);
  position: relative;
  overflow-y: hidden;
  ${bgTransition}
`
