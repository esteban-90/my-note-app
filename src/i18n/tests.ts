import type { Resource } from 'i18next'
import { createInstance } from 'i18next'
import { initReactI18next } from 'react-i18next'
import { ns, supportedLngs } from './app'

const resources: Resource = {}
export const i18nInstance = createInstance()

void (async () => {
  for (const n of ns) {
    for (const lng of supportedLngs) {
      resources[lng] ??= {}
      resources[lng] = {
        ...resources[lng],
        [n]: await import(`../../public/translations/${lng}/${n}.json`),
      }
    }
  }
})()

i18nInstance.use(initReactI18next).init({
  ns,
  supportedLngs,
  fallbackLng: supportedLngs[0],
  lng: supportedLngs[0],
  react: { useSuspense: false },
  resources,
})
