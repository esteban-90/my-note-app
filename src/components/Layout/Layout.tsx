import type { FC, PropsWithChildren } from 'react'
import { useTheme } from '@/contexts'
import { Wrapper, Content } from '@styled/Layout'

export const Layout: FC<PropsWithChildren> = ({ children }) => {
  const { theme } = useTheme()

  return (
    <Wrapper className={theme} data-testid='app-layout'>
      <Content>{children}</Content>
    </Wrapper>
  )
}
