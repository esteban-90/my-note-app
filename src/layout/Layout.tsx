import type { FC, PropsWithChildren } from 'react'
import { Wrapper, Content } from './Layout.styled'

export const Layout: FC<PropsWithChildren> = ({ children }) => {
  return (
    <Wrapper data-testid='app-layout'>
      <Content>{children}</Content>
    </Wrapper>
  )
}
