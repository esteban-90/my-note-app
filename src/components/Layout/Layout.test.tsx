import { screen, render } from '@/tests'
import { Layout } from './Layout'

describe(Layout.name, () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    ;({ asFragment } = render(<Layout />))
  })

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it(`should render ${Layout.name} component`, () => {
    const layoutComponent = screen.getByTestId('app-layout')
    expect(layoutComponent).toBeInTheDocument()
  })
})
