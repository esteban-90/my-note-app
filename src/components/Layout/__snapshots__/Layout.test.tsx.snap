// Vitest Snapshot v1

exports[`Layout > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="night css-10j4kk1"
    data-testid="app-layout"
  >
    <div
      class="css-14g8731"
    />
  </div>
</DocumentFragment>
`;
